### ISSUE TITLE
Release schedule for Brat frontend

### ISSUE THREAD

**Issue by victoryhb** [crim-ca/brat-frontend-editor#3]
Hi Renaud,
I am a research student from Hong Kong working on a Web-based annotator in one of our projects. I and my partner are currently looking into using Brat as our main interface and came across your forked front-end, which we have tested and are quite excited about. 
We see the front-end still has quite a few rough edges and wonder if a new release will be coming soon. Also we're interested in contributing to the project if possible (although we only have very limited experience with open-sourced projects) and so please let us know if we could be of help in any way.

Victor

**Comment by Renaud009** [comment]
Hello Victor,
We definitly are going to spend number of hours on this project sooner than later. I have spent a few days just to figure out if it could actually be done easily (current master). We have a few students that are probably going to spent a couple days on this project very soon (1-2 weeks) then I plan to spend like 15 days more days if needed in early september to have a stable version ready for release.

Our organisation is also starting in the open source world and in building open communities so we are more than happy if you are willing to contribute. I just commited a CONTRIBUTING.md file based on the main brat file which perfectly fits our needs, more info here: https://github.com/crim-ca/brat-frontend-editor/blob/master/CONTRIBUTING.md

Could you point me the "few rough edges" you talked about so we can start logging a few issues (I have also a few ideas on what needs to be fixed or implemented, haven't had time to had them as github issues). I have no problem if you guys start opening issues and pull-request branches to fix obvious bugs.

Keep me in touch

Renaud

**Comment by victoryhb** [comment]
Hey Renaud,
Thank you for the encouraging words and we look forward to the coming releases. As we test out the editor we'll document bugs and missing features in github issues. Hopefully we could contribute our little bit to the project along the way.

Thanks,
Victor

**Comment by reckart** [comment]
Do you plan for any strategy of keeping brat and this project in sync? Or will we end up with a true fork?

**Comment by Renaud009** [comment]
@victoryhb Awesome!

@reckart 
We could definitly try to be in sync with the main brat project (or at least the frontend files) by using the notion of git upstreams or some other mechamisms (I am not too familiar with these tho). 

At this point, the main concern probably is to limit the modifications of common files. I think it's kinda possible (thanks to the dispatch pattern they used) with some limitations. I guess we could eventually try to merge with their master when we'll have a more stable version. But will they accept or not the pull-request is out of my control.

**Comment by reckart** [comment]
@Renaud009 in WebAnno, we also had to make several modifications. Mainly, we modified visualize.js and in our version, we tried to mark very clearly which section we changed, trying to always keep the original sections as well commented out. It has been a while since we synched with brat, but as far as I know they didn't make significant changes in that file since. Although, I didn't check since the brat folks merged back our changes for right-to-left languages ;)

**Comment by Renaud009** [comment]
@reckart Yes I actually looked at the way your guys from WebAnno integrated their files before starting my own fork. Every line I changed also comes with a comment :-)
